game/scripts/vscripts/items/item_golden_arrow.lua

```diff
@@ -101,8 +32,69 @@ function modifier_item_golden_arrow:GetModifierMoveSpeedBonus_Constant()
 end
 
 modifier_item_golden_arrow_counter = class({
-	GetTexture = function() return "item_arena/golden_arrow_attributes" end,
+	GetTexture = function() return "item_arena/golden_arrow" end,
 	IsHidden = function() return false end,
 	IsPurgable = function() return false end,
-	RemoveOnDeath = function() return false end,
 })
+
+
+modifier_item_golden_arrow_target = class({
+	IsPurgable = function() return true end,
+	IsHidden = function() return false end,
+})
+
+
+if IsServer() then
+	function item_golden_arrow:OnSpellStart()
+		local caster = self:GetCaster()
+		local target = self:GetCursorTarget()
+		if caster:GetLevel()>= self:GetSpecialValueFor("max_caster_level") then return end
+		target:AddNewModifier(caster, self, "modifier_item_golden_arrow_target", {duration = self:GetSpecialValueFor("duration")})
+	end
+
+	function modifier_item_golden_arrow_target:OnCreated()
+		self:StartIntervalThink(0.6)
+	end
+
+	function modifier_item_golden_arrow_target:OnIntervalThink()
+		local caster = self:GetCaster()
+		local particle = ParticleManager:CreateParticle("particles/arena/items_fx/golden_arrow_target_b.vpcf", PATTACH_ABSORIGIN_FOLLOW, self:GetParent())
```

Comment:
So, you're recreating particle each 0.6 seconds? That should be done in particle itself